.github/workflows/main.yml

```diff
@@ -80,7 +80,16 @@ jobs:
 
     # Deny warnings on CI to avoid warnings getting into the codebase.
     - run: cargo test --features 'deny-warnings'
-    # The testsuite generates a huge amount of data, and fetch-smoke-test was
+    - name: Clear intermediate test output
+      run: |
+          df -h
+          rm -rf target/tmp
+          df -h
+    - name: gitoxide tests (all git-related tests)
+      run: cargo test git
+      env:
+        RUSTFLAGS: "--cfg always_test_gitoxide"
```

Comment:
I'd like to see a different option other than using `--cfg`. This is pretty expensive to rebuild everything. Can you say more about why an environment variable wasn't working for you? We use it elsewhere for testing, such as `CARGO_RUN_BUILD_STD_TESTS`.

Reply:
Something to note is that this controls production code iirc.  If we used an env variable, we'd want to name it appropriately as a test-only variable.

Reply:
Thanks for pointing it out! It felt a bit like a hack to make it work and let me focus on the important bits, and I left it there even though I also believe environment variables should work.

I will try again with environment variables and see if there is anything I did wrongly by comparing it to `CARGO_RUN_BUILD_STD_TESTS`, and should it not work I will share it here so we can sort it out together. TODO list updated.